tools/build-templated-pages/src/features.rs

```diff
@@ -58,53 +38,88 @@ fn parse_features(panic_on_missing: bool) -> Vec<Feature> {
+                    if let Some(description) = key_prefix {
+                        create_feature(&default, key_name, description)
+                    } else {
+                        panic!("Missing description for feature {key_name}");
                     }
-                    let description = description
-                        .strip_prefix("\n# ")
-                        .unwrap()
-                        .strip_suffix('\n')
-                        .unwrap()
-                        .to_string();
-                    Some(Feature {
-                        is_default: default.contains(&name),
-                        name,
-                        description,
-                    })
-                } else if panic_on_missing {
-                    panic!("Missing description for feature {name}");
                 } else {
-                    None
+                    let feature = |description| create_feature(&default, key_name, description);
+                    key_prefix.map(feature).unwrap()
                 }
             }
         })
         .collect()
 }
 
+fn get_manifest_features() -> Table {
+    let manifest_file = std::fs::read_to_string("Cargo.toml").unwrap();
+    let manifest = manifest_file.parse::<Document>().unwrap();
+    let features = manifest["features"].as_table().unwrap().clone();
+    features
+}
+
+fn create_feature(default: &[String], name: String, description: &RawString) -> Option<Feature> {
+    let description = description.as_str().unwrap().to_string();
+    if let Some(stripped_description) = description
+        .strip_prefix("\n# ")
+        .and_then(|d| d.strip_suffix('\n'))
+    {
+        let is_default = default.contains(&name);
+        let description = get_description(stripped_description.to_string(), &name);
+        Some(Feature {
+            is_default,
+            name,
+            description,
+        })
+    } else {
+        panic!("Missing description for feature {name}");
+    }
+}
+
+fn get_description(description: String, name: &String) -> String {
+    let description = description.to_string();
+    if !description.starts_with("\n# ") || !description.ends_with('\n') {
+        panic!("Missing description for feature {name}");
+    }
+    let stripped_description = description.strip_prefix("\n# ").unwrap().strip_suffix('\n');
+    stripped_description.unwrap().to_string()
+}
+
+fn get_key_name(key: &Key) -> String {
+    let key_repr = key.as_repr().unwrap();
+    key_repr.as_raw().as_str().unwrap().to_string()
+}
+
+fn get_default_features(features: &Table) -> Vec<String> {
+    let value = |v: &Value| {
+        let feature_name = v.as_str().unwrap();
+        let feature = |v: &Value| v.as_str().unwrap().to_string();
+        let features_to_array = |name| features.get(name).unwrap().as_array().unwrap();
+        let map = features_to_array(feature_name).iter().map(feature);
+        std::iter::once(feature_name.to_string()).chain(map)
+    };
```

Comment:
This is quite difficult to read. Is there a way to refactor this so it is more understandable? (Even better, if you can return `impl Iterator<Item = String / &str>`.)

If that's not possible, could you add comments explaining the purpose of this code?